### DISCUSSION TITLE
Hide the soft keyboard of an entry?

### DISCUSSION THREAD

**Discussion by stroetmann-schattke** [dotnet/maui#16061] [Q&A]
Hello,

I am working on an app for Android which lets the user scan barcodes.
These barcodes with their quantities are stored into a local db for later use.

The scanner is build into the device and uses the keyboard buffer.
I have modified the Android MainActivity to have an OnPageKeyDown-Event where I call an Entry.Focus();

Unfortunately, I am struggling for some days now being unable to hide the soft keyboard.
I tried several things and failed with everything:

1) Handled the Focus-Event and called:
   Entry.IsEnabled = false; Entry.IsEnabled = true;
   Problem: it does hide the keyboard, but the scanner is no longer able to enter the keystrokes into the entry anymore.

2) I created a class NoKeyboardEntry with its own EntryRenderer, setting the
   this.Control.ShowSoftInputOnFocus = false;
   and calling
   InputMethodManager.HideSoftInputFromWindow on IsFocused PropertyChanging.
   Problems: it only works 50% as there is no soft keyboard when touching the entry,
   but still pops up when the page shows up and when I focus the entry programmatically.
   Also, the Android physical back-button is no longer working correctly.

3) I added [Activity(... WindowSoftInputMode = Android.Views.SoftInput.StateAlwaysHidden, ...] to the Android MainActitivy.
   Also I added Window.SetSoftInputMode(Android.Views.SoftInput.StateAlwaysHidden); on OnCreate.
   No effect.

4) I tried to call this code on creating the page:
   Microsoft.Maui.Handlers.EntryHandler.Mapper.AppendToMapping(...
      ... handler.PlatformView.InputType = Android.Text.InputTypes.Null;
   No effect.

5) I wanted to give the Xamarin.CommunityToolkit a try since there seem to be some keyboard hide helpers,
   but I get a wall of compatibility errors when trying to install :-(

Currently, the soft keyboard constantly is popping up and down again every second when I scan barcodes.
This solution is very clumsy. Why is there no simple keyboard="none" possibility!?

I use the following code to hide the soft keyboard:
Platform.CurrentActivity.HideKeyboard(Platform.CurrentActivity.CurrentFocus);
which is called over and over again since you apparently cannot find out if the soft keyboard is visible or not.

Do you have any ideas or suggestions?

Thanks,
Mirko

**Comment by acaliaro** [comment]
HI. For "enterprise" type devices, with the integrated scanner, I always suggest not to use the keyboard emulation to read the barcodes. Usually the major vendors give the possibility to use broadcasts with intents to retrieve the barcode. What device are you using?

**Reply by stroetmann-schattke** [reply]
We have 2 different ones:
ALMEX TC601B
SEUIC Cruise-1

**Comment by stroetmann-schattke** [comment]
I added a video of my app:

https://github.com/dotnet/maui/assets/63344665/4aeb4001-c3dd-4e26-beab-dbe64a4d8445

**Comment by acaliaro** [comment]
for almex you can define the broadcastreceiver
https://www.imotiongroup.com/wp-content/uploads/TC601-Manual_EN_20200416_compressed-1.pdf page 33
for seuic
https://www.mobileworxs.com/wp-content/uploads/2018/09/Cruise-1-7.1-manual-MWX.pdf page 50

they talk about broadcast...

you can take a look to my post
https://acaliaro.wordpress.com/2023/06/08/read-a-barcode-with-zebra-terminals-using-datawedge-and-dotnetmaui/

to understand how to use broadcastreceiver to read barcodes

**Reply by stroetmann-schattke** [reply]
Thank you, Alessandro,
I will give that a try and will make it a configurable option.

Do you know if it is possible to turn off the soft keyboard completely? ;-)
